tests/functional/api/test_gitlab.py

```diff
@@ -23,6 +24,24 @@ def test_auth_from_config(gl, temp_dir):
     assert isinstance(test_gitlab.user, gitlab.v4.objects.CurrentUser)
 
 
+def test_no_custom_session(gl, temp_dir):
+    """Test no custom session"""
+    custom_session = requests.Session()
+    test_gitlab = gitlab.Gitlab.from_config(
+        config_files=[temp_dir / "python-gitlab.cfg"]
+    )
+    assert test_gitlab.session != custom_session
+
+
+def test_custom_session(gl, temp_dir):
+    """Test custom session"""
+    custom_session = requests.Session()
+    test_gitlab = gitlab.Gitlab.from_config(
+        config_files=[temp_dir / "python-gitlab.cfg"], session=custom_session
+    )
+    assert test_gitlab.session == custom_session
```

Comment:
Since we're not making any real requests, I think these could be in unit tests, and done on `gitlab.Gitlab()` where the session logic is handled.

Reply:
Could you please clarify how a real request can make a difference? 
I could add unit tests as well if you think it makes sense.

Reply:
Sorry for the confusion, what I meant to say is that here we're essentially testing the `__init__()` method of the `Gitlab` class, so ideally we would put these tests in https://github.com/python-gitlab/python-gitlab/blob/main/tests/unit/test_gitlab.py. Probably somewhere in here:

https://github.com/python-gitlab/python-gitlab/blob/0ecf3bbe28c92fd26a7d132bf7f5ae9481cbad30/tests/unit/test_gitlab.py#L76-L96

Reply:
I've tried the direction you are suggestion. It complicates the implementation while I don't fully understand the value. In my point of view, testing the `__init__` function functionality is valuable.

Reply:
After re-reviewing the tests, those functions tests the `from_config` function

Reply:
I've decided to implement both unit tests and functional tests. It might be redundant for now but still worth keeping both of them.

Reply:
Thanks! I think this can go in as is, but just to clarify: my point regarding unit/functional was more that this test can easily be done in isolation and doesn't need to live in the functional test suite where we fire up a real server to make requests against :)

I guess there was some confusion about `__init__`, my suggestion would be to simply change the test to assert after `gitlab.Gitlab()` is instantiated with a custom session, as that is where the logic is. In any case, https://github.com/python-gitlab/python-gitlab/pull/2397 should help in the future, as I assume you are testing `from_config` mostly because we use it everywhere in our functional tests.

Reply:
TBH, I'm not following :-)